libraries/rush-lib/src/utilities/SetRushLibPath.ts

```diff
@@ -2,11 +2,9 @@ import { PackageJsonLookup } from '@rushstack/node-core-library';
 
 import { EnvironmentVariableNames } from '../api/EnvironmentConfiguration';
 
-if (!process.env[EnvironmentVariableNames.RUSH_LIB_PATH]) {
-  const rootDir: string | undefined = PackageJsonLookup.instance.tryGetPackageFolderFor(__dirname);
-  if (rootDir) {
-    // Route to the 'main' field of package.json
-    const rushLibIndex: string = require.resolve(rootDir, { paths: [] });
-    process.env[EnvironmentVariableNames.RUSH_LIB_PATH] = rushLibIndex;
-  }
+const rootDir: string | undefined = PackageJsonLookup.instance.tryGetPackageFolderFor(__dirname);
+if (rootDir) {
+  // Route to the 'main' field of package.json
+  const rushLibIndex: string = require.resolve(rootDir, { paths: [] });
+  process.env[EnvironmentVariableNames.RUSH_LIB_PATH] = rushLibIndex;
```

Comment:
👍 I think this was actually the originally intended design.

Reply:
I don't think I'd fully registered that the version selector was hitting the code path that defined the variable, and expected it to get set only when it loaded the underlying library.

Reply:
What happens if `rushVersion` is significantly /older/ than the globally installed Rush?

Reply:
if it's before rush-sdk got the env variable feature, it'll work fine because it doesn't have this issue. If it has the env var thing, it'll continue to be broken until they upgrade.